src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -47,8 +53,47 @@ bool BinaryCompactObject::Object::is_excised() const noexcept {
   return inner_boundary_condition.has_value();
 }
 
-void BinaryCompactObject::check_for_parse_errors(
```

Comment:
refactoring these constructors is also something where doing some of the changes in a separate commit would make life easier.